pathd/path_pcep_cli.c

```diff
@@ -0,0 +1,1480 @@
+	} else {
+		return CMD_ERR_NO_MATCH;
+	}
+
+	strncpy(config_group->tcp_md5_auth, argv[1]->arg, TCP_MD5SIG_MAXKEYLEN);
+
+	return CMD_SUCCESS;
+}
+
+static int path_pcep_cli_peer_address(struct vty *vty, int argc,
+				      struct cmd_token *argv[])
+{
+	/* There must be either 3 or 5 arguments */
+	if (argc < 3 || argc == 4 || argc > 5) {
+		return CMD_ERR_INCOMPLETE;
+	}
+
+	struct pce_opts *pce_opts = NULL;
+	if (vty->node == PCC_PEER_NODE) {
+		/* TODO need to see if the pce is in use, and reset the
+		 * connection */
+		pce_opts = current_pce_opts_g;
+		current_pce_opts_g->merged = false;
+	} else {
+		return CMD_ERR_NO_MATCH;
+	}
+
+	if (strcmp(PCEP_VTYSH_ARG_IPV6, argv[1]->arg) == 0) {
+		pce_opts->addr.ipa_type = IPADDR_V6;
+		if (!inet_pton(AF_INET6, argv[2]->arg,
+			       &pce_opts->addr.ipaddr_v6)) {
+			return CMD_ERR_INCOMPLETE;
+		}
+	} else if (strcmp(PCEP_VTYSH_ARG_IP, argv[1]->arg) == 0) {
+		pce_opts->addr.ipa_type = IPADDR_V4;
+		if (!inet_pton(AF_INET, argv[2]->arg,
+			       &pce_opts->addr.ipaddr_v4)) {
+			return CMD_ERR_INCOMPLETE;
+		}
+	} else {
+		return CMD_ERR_NO_MATCH;
+	}
+
+	/* Handle the optional port */
+	int i = 3;
+	int port = PCEP_DEFAULT_PORT;
+	if (argc == 5) {
+		if (strcmp(PCEP_VTYSH_ARG_PORT, argv[i]->arg) == 0) {
+			PCEP_VTYSH_INT_ARG_CHECK(i, port, 0, 65535);
+		} else {
+			return CMD_ERR_NO_MATCH;
+		}
+	}
+	pce_opts->port = port;
+
+	return CMD_SUCCESS;
+}
+
+static int path_pcep_cli_peer_pcep_config_group(struct vty *vty, int argc,
+						struct cmd_token *argv[])
+{
+	/* There must be 2 arguments */
+	if (argc != 2) {
+		return CMD_ERR_INCOMPLETE;
+	}
+
+	struct pce_opts *pce_opts = NULL;
+	if (vty->node == PCC_PEER_NODE) {
+		/* TODO need to see if the pce is in use, and reset the
+		 * connection */
+		pce_opts = current_pce_opts_g;
+		current_pce_opts_g->merged = false;
+	} else {
+		return CMD_ERR_NO_MATCH;
+	}
+
+	struct pcep_config_group_opts *config_group =
+		pcep_cli_find_pcep_config_group(argv[1]->arg);
+	if (config_group == NULL) {
+		vty_out(vty, "pce-config-group [%s] does not exist.\n",
+			argv[1]->arg);
+		return CMD_ERR_INCOMPLETE;
+	}
+
+	strcpy(pce_opts->config_group_name, config_group->name);
+
+	return CMD_SUCCESS;
+}
+
+static int path_pcep_cli_peer_timers(struct vty *vty, int argc,
+				     struct cmd_token *argv[])
+{
+	/* There must be at least 3 arguments */
+	if (argc < 3) {
+		return CMD_ERR_INCOMPLETE;
+	}
+
+	struct pcep_config_group_opts *config_group = NULL;
+	if (vty->node == PCC_PEER_NODE) {
+		/* TODO need to see if the pce is in use, and reset the
+		 * connection */
+		config_group = &current_pce_opts_g->config_opts;
+		current_pce_opts_g->merged = false;
+	} else if (vty->node == PCEP_CONFIG_GROUP_NODE) {
+		config_group = current_pcep_config_group_opts_g;
+	} else {
+		return CMD_ERR_NO_MATCH;
+	}
+
+	/* Handle the arguments */
+	int i = 1;
+	while (i < argc) {
+		if (strcmp(PCEP_VTYSH_ARG_KEEP_ALIVE, argv[i]->arg) == 0) {
```

Comment:
You should probably use `argv_find` like at https://github.com/opensourcerouting/frr/blob/master/isisd/isis_redist.c#L558

We are trying to stop using strcmp and hardcoded index.